extensions/gitpod-remote/src/extension.ts

```diff
@@ -16,7 +18,11 @@ export async function activate(context: vscode.ExtensionContext) {
 		return;
 	}
 
+	const workspaceInfo = await gitpodContext.gitpod.server.getWorkspace(gitpodContext.info.getWorkspaceId());
+	const wsConfig = workspaceInfo.workspace.config;
+
 	registerTasks(gitpodContext);
+	configureMachineSettings(gitpodContext, wsConfig).then(() => tunnelInitialPorts(gitpodContext!, workspaceInfo.workspace.config.ports ?? []));
```

Comment:
`workspaceInfo.workspace.config` is for supervisor, it already respects it, clients should not care about them: https://github.com/gitpod-io/gitpod/blob/39c5542024f7717041efde80cd1d5476e725ca7b/components/supervisor/pkg/ports/ports-config.go#L123-L128

[[1](https://github.com/gitpod-io/gitpod/blob/39c5542024f7717041efde80cd1d5476e725ca7b/components/supervisor-api/status.proto#L56)] is API fo clients, it already factors in all possible sources